### PR TITLE
fix: pass PermissionGate after granting permission in Settings and returning to app

### LINKED ISSUE #1206: Granting camera permission from Settings requires an app restart in Android
**Describe the bug**
If you deny camera permissions, the app prompts you to open the Settings app to grant the permission. If you do so and return to the app, it still asks you to do the same thing.

**To Reproduce**
1. Fresh install of the app (or deny camera permission in Settings)
2. Enter the StandardCamera
3. You should see the camera permission gate
4. Tap OBSERVE ORGANISMS
5. Tap "Don't allow"
6. Tap OPEN SETTINGS
7. In the Settings app, grant camera permission (tap Permissions, tap Camera, tap "Allow only while using the app"
8. Return to iNat (I usually use the app switcher)
9. The permission gate still displays and is prompting you to open settings... even though you just did that
10. Tap the x button in the upper right to close
11. Enter the StandardCamera again
12. It's still asking you to open settings

**Expected behavior**
Once you've granted camera permission in Settings, iNat should realize that, e.g. when the permission gate has focus, it should check for the permission again.

**Context (please complete the following information):**
 - Device: Pixel 8
 - OS: Android 14
 - App version: 0.23.0 (72)

**Additional context**
Works on iOS, so be careful that any changes don't break this there.

### PR DESCRIPTION
**Pull request by israr002**
closes #1206 

https://github.com/user-attachments/assets/3e5bddc1-4477-4906-8a0d-2e52e53ac40c

### CODE REVIEW

File: src/components/SharedComponents/PermissionGateContainer.tsx [hunk 1]
```diff
@@ -204,6 +206,32 @@ const PermissionGateContainer = ( {
     }
   }, [result, children, setModalShown] );
 
+  useEffect( () => {
+  // We need to handle permission changes manually on Android
+  // Permissions modified from the settings are not captured automatically
+  // So we check permissions when the app is transitioned from background to foreground
+  // This check is performed only when the platform is Android and the permission result is BLOCKED
+    if ( Platform.OS === "android"
+       && ( result === RESULTS.BLOCKED ) ) {
```
Comment (kueda) [inline src/components/SharedComponents/PermissionGateContainer.tsx:215]: Unnecessary parens here.

File: src/components/SharedComponents/PermissionGateContainer.tsx [hunk 2]
```diff
@@ -204,6 +206,32 @@ const PermissionGateContainer = ( {
...
+
+      const subscription = AppState.addEventListener( "change", onAppStateChange );
+
+      return () => {
+        subscription?.remove();
+      };
+    }
+
+    return () => undefined;
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (kueda) [inline src/components/SharedComponents/PermissionGateContainer.tsx:232]: Why are you ignoring the exhaustive deps warning?

**Review by kueda — CHANGES_REQUESTED** [review]
This works well, but I'm not sure why you're disabling the exhaustive deps warning. In general if you disable an linting rule, please explain why in a comment.

File: src/components/SharedComponents/PermissionGateContainer.tsx (hunk 1 above)
Comment (israr002) [reply src/components/SharedComponents/PermissionGateContainer.tsx]: removed

File: src/components/SharedComponents/PermissionGateContainer.tsx (hunk 2 above)
Comment (israr002) [reply src/components/SharedComponents/PermissionGateContainer.tsx]: added checkPermission dependency

**Review by kueda — APPROVED** [review]

**Pull request merged**